clients/client/src/main/java/org/projectnessie/client/http/OpentelemetryTracing.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.client.http;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.io.IOException;
+import org.projectnessie.api.NessieVersion;
+
+final class OpentelemetryTracing {
+  private OpentelemetryTracing() {}
+
+  static void addTracing(HttpClient.Builder httpClient) {
+    // It's safe to reference `GlobalTracer` here even without the required dependencies available
+    // at runtime, as long as tracing is not enabled. I.e. as long as tracing is not enabled, this
+    // method will not be called and the JVM won't try to load + initialize `GlobalTracer`.
+    OpenTelemetry otel = GlobalOpenTelemetry.get();
+    Tracer tracer = otel.getTracer("Nessie");
+    if (tracer != null) {
+      httpClient.addRequestFilter(
+          context -> {
+            Span span =
+                tracer
+                    .spanBuilder("Nessie-HTTP")
+                    .setSpanKind(SpanKind.CLIENT)
+                    .startSpan()
+                    .setAttribute(SemanticAttributes.HTTP_URL, context.getUri().toString())
+                    .setAttribute(SemanticAttributes.HTTP_METHOD, context.getMethod().name())
+                    .setAttribute("nessie.version", NessieVersion.NESSIE_VERSION);
+
+            W3CTraceContextPropagator.getInstance()
```

Comment:
I just realized that we should probably take the propagator from the env.... that is use whatever is configured in the "container" running the Nessie client... I do not recall off the top of my head what API we need to use on the OTel side, but I hope something is available to let us get "active" propagators :thinking:

Reply:
We had the Jaeger specific one before, but that seems to have disappeared, so there's the W3C one here.

I'd assume, that if tracing is explicitly enabled, it's properly setup in the calling code?